ibis/common/patterns.py

```diff
@@ -821,6 +798,15 @@ class InstanceOf(Slotted, Singleton, Pattern):
     def __init__(self, typ):
         super().__init__(type=typ)
 
+    def describe(self, plural=False):
+        typename = self.type.__name__
+        if plural:
+            return f"{self.type.__name__}s"
+        elif typename[0].lower() in "aeiou":
+            return f"an {self.type.__name__}"
+        else:
+            return f"a {self.type.__name__}"
```

Comment:
Great attention to detail 🧐

Reply:
I'm not sure how well will it compose with nested patterns. I still need to implement describe() for at least the common ones AnyOf, AllOf, GenericInstanceOf and GenericCoercedTo